docs/root/faq/rolling_upgrades.rst

```diff
@@ -0,0 +1,124 @@
+at the network level.
+
+Retries
+-------
+
+Automatic :ref:`request retries<config_http_filters_router>` is another measure of ensuring service resilience. Request retries should typically be used to guard against 
+transient failures. Envoy supports very rich set of configurable parameters that dictate what type of requests are retried, how many times 
+the request should be retried, timeouts for retries, etc.
+
+Retries in gRPC services
+------------------------
+
+For gRPC services, Envoy looks at the gRPC status in the response and attempts a retry based on the statuses configured in *x-retry-grpc-on*.
+
+The following application status codes in gRPC are considered safe for automatic retry.
+
+* *CANCELLED* - Return this code if there is an error that can be retried in the service.
+* *RESOURCE_EXHAUSTED* - Return this code if some of the resources that service depends on are exhausted in that instance so that retrying 
+  to another instance would help. Please note that for shared resource exhaustion, returning this will not help. Instead :ref:`rate limiting<arch_overview_rate_limit>`
+  should be used to handle such cases.
+
+The Http Status codes *502 (Bad Gateway)*, *503 (Service Unavailable)* and *504 (Gateway Timeout)* are all mapped to gRPC status code *UNAVAILABLE*. 
+This can also be considered safe for automatic retry.
+
+An important consideration should be given to *idempotency* of a request while configuring retries.
+
+Envoy also supports various extensions to the standard retry policy supported out of the box. 
+The :ref:`retry plugins<arch_overview_http_retry_plugins>` allow you to customize the Envoy 
+retry implementation to your application.
+
+Outlier Detection
+-----------------
+
+:ref:`Outlier detection<arch_overview_outlier_detection>` is a way of dynamically determining outlier 
+hosts in the upstream cluster. If certain hosts behave differently from others like returning high 5xx 
+or success rate differs  from other hosts, etc. By detecting such hosts and ejecting them for a 
+temporary period of time from the healthy load balancing set, Envoy can increase the success rate of a request. 
+Envoy supports configuring outlier detection based on continuous 5xx, continuous gateway failures and success rate.
+
+Envoy also allows you to configure the ejection period.
+
+**Configuration**
+
+The following settings help to optimize some combination of:
+
+* Maximum request success for common scenarios (i.e. rolling upgrade) 
+* Speed 
+* Avoid cascading failures
+
+
+*Circuit Breaker*
+
+.. code-block:: json
+
+  {
+     "thresholds": [
+       {
+         "max_retries": 10,
+       }
+    ]
+  }
+
+For the purpose of this specific use case, the retry budget for upstream cluster should be configured to
+enable and control concurrent retries. If the value configured is too low, some requests will not be retried 
+and will result in :ref:`upstream_rq_retry_overflow <config_cluster_manager_cluster_stats>`
```

Comment:
Replace "... retried and will result in" with "retried, which can be measure via...